pkgs/tools/archivers/7zz/default.nix

```diff
@@ -50,6 +54,10 @@ stdenv.mkDerivation rec {
   sourceRoot = "CPP/7zip/Bundles/Alone2";
 
   makeFlags =
+    [
+      "CC=${stdenv.cc.targetPrefix}cc"
```

Comment:
There's a `CROSS_COMPILE` makeflag/var

I guess we could do `"CROSS_COMPILE=${stdenv.cc.targetPrefix}"`? but then we might also want to turn it on only for cross compiling?

Reply:
I don't think we need. Probably `CROSS_COMPILE` is for a specific need for cross compilation, but just setting `CC` and `CXX` seems sufficient to get a working cross compile build for me.

And why set this regardless variables regardless? This is considered best practice in `nixpkgs`. Also, there is no issue of this causing our build to use a compiler we don't expect (like in other systems), thanks to the Nix hermetic build system.

Reply:
BTW, if `CROSS_COMPILE` adds the prefix to the compiler, setting it alone wouldn't work. We would also need to set both `CC=cc` and `CXX=c++` since we can't assume that the C/C++ compiler will always be GCC. That is, unless the package only builds with GCC, them this is probably a better solution, but AFAIK `7zz` builds happily with Clang (at least the generic build).

Reply:
Looks like it's just for prefix

```
λ rg CROSS_COMPILE ../../..
../../../7zip/var_clang.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_clang_x64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_clang_arm64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_clang_x86.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_gcc_x64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)gcc
10:CXX=$(CROSS_COMPILE)g++

../../../7zip/var_mac_arm64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_gcc_x86.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)gcc
10:CXX=$(CROSS_COMPILE)g++

../../../7zip/var_mac_x64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)clang
10:CXX=$(CROSS_COMPILE)clang++

../../../7zip/var_gcc.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)gcc
10:CXX=$(CROSS_COMPILE)g++

../../../7zip/var_gcc_arm64.mak
6:CROSS_COMPILE=
9:CC=$(CROSS_COMPILE)gcc
10:CXX=$(CROSS_COMPILE)g++
```

---

Is there a reason `"CC=${stdenv.cc.targetPrefix}cc"` etc isn't just default stdenv behavior in nixpkgs if it's used so often?

Reply:
> Is there a reason "CC=${stdenv.cc.targetPrefix}cc" etc isn't just default stdenv behavior in nixpkgs if it's used so often?

No idea.

I would assume that this was something that they eventually decided it was a good idea, but turning by default would break tons of things. But I am only assuming this, I have no idea if this is true or not.

Reply:
> I guess we could do "CROSS_COMPILE=${stdenv.cc.targetPrefix}"? but then we might also want to turn it on only for cross compiling?

If you do want to use it instead for some reason (since it has this flag maybe upstream may use it for something else in the future), I think we can do this change. No need to set it conditionally though, if there is no cross compilation happening `stdenv.cc.targetPrefix` will be set to empty string.

Reply:
ok cool. well it looks like `CROSS_COMPILE` would be the 7zz maintainer's "preferred" way of adding this but it's much more nixpkgs standard to have it this way

Although as a note there are a handful of packages using `CROSS_COMPILE`

```
λ rg '"CC=\$\{stdenv\.cc\.targetPrefix\}cc"' | wc -l
111
λ rg '"CROSS_COMPILE=\$\{stdenv\.cc\.targetPrefix\}"' | wc -l
15
```